pq-crypto/s2n_kyber_512_evp.h

```diff
@@ -0,0 +1,28 @@
+/*
+* Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License").
+* You may not use this file except in compliance with the License.
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/
+
+#pragma once
+
+#include <openssl/evp.h>
+
+#include "tls/s2n_kem.h"
+
+#if (defined(S2N_AWSLC_KYBER) && defined(OPENSSL_IS_AWSLC) && AWSLC_API_VERSION >= 20)
```

Comment:
Is there a way to make this a feature probe instead of a version check?

Reply:
Should be possible, but would like to know a little more around the motivation. Are you thinking along the lines of a compile-time or run-time feature probe?  What would that strategy get us? Do you have an example of the kind of pattern you're thinking about?

Reply:
Compile time probe. The motivation is avoiding fragile lists of checks and simply saying if a file compiles then it supports the feature. We have many examples of doing this: https://github.com/aws/s2n-tls/tree/main/tests/features
https://github.com/aws/s2n-tls/blob/2e67a8af55ab7bd1d4601708fa47c642de3c9bfb/CMakeLists.txt#L163-L174

Reply:
https://github.com/aws/s2n-tls/issues/3327

Reply:
Would you block merging on this change? AFAICT, the API version check approach here is the one that was proposed and agreed on by the PQ, S2N, and internal load balancer teams.